llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -58,7 +65,13 @@ instance Monad m => EncodeM m A.FA.FunctionAttribute (Ptr FFI.FunctionAttrBuilde
     liftIO $ FFI.attrBuilderAddStringAttribute b kindP kindLen valueP valueLen
   encodeM a = return $ \b -> liftIO $ case a of
     A.FA.StackAlignment v -> FFI.attrBuilderAddStackAlignment b v
+    A.FA.AllocSize x y -> case y of
+                            Nothing -> FFI.attrBuilderAddAllocSize b x 0 0
+                            Just y' -> FFI.attrBuilderAddAllocSize b x 1 y'
```

Comment:
I need a case somewhere, I can either leave it here or push it in the FFI directory but using a tuple doesn’t change a fact that at some point I need to case on the maybe.